src/systems/filecoin_nodes/clock/clock_subsystem.go

```diff
@@ -2,10 +2,27 @@ package clock
 
 import "time"
 
-func (_ *WallClock_I) NowUTC() Time {
+func (_ *Clock_I) UTCTime() Time {
 	return Time(time.Now().Unix())
 }
 
-func (_ *WallClock_I) NowUTCNano() Time {
+func (_ *Clock_I) UnixNano() Time {
 	return Time(time.Now().UnixNano())
 }
+
+func (self *Clock_I) CurrentEpoch() ChainEpoch {
+	return self.currentEpoch_
+}
+
+func (self *Clock_I) CurrentEpochState() ChainEpoch {
+	panic("")
+	// if self.UnixNano()-self.currentEpochStart > self.cutoffAt_ {
+	// 	return ChainEpochState.PastCutoff
+	// }
+	// return ChainEpochState.Active
+}
```

Comment:
remove -- dont hold state

Reply:
ok, however we do need a way to enforce "when" in the epoch we are as part of rejecting blocks received past cutoff in block validation... Will change so it's stateless.